dali/operators/audio/nonsilence_op.h

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_AUDIO_NONSILENCE_OP_H_
+#define DALI_OPERATORS_AUDIO_NONSILENCE_OP_H_
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <utility>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+#include "dali/kernels/signal/moving_mean_square.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+const int kNumOutputs = 2;
+
+template<typename InputType>
+struct Args {
+  TensorView<StorageCPU, const InputType, 1> input;
```

Comment:
I think it'd be better to remove `input` from here (and get rid of the template type)

Reply:
As we discussed it while ago, in DALI (especially in kernels) input and args are treated almost the same way. Is there particular reason not to do so here? I don't see any major problems that might occur by making input one of the args. After all, both input and args are the coefficient, based on which output is calculated

Reply:
There is a difference between arguments describing the nature of the transformation and the input data you apply the transformation on.

On top of that, by keeping `input` here you need to have this struct templatized which makes the code unnecessarily complicated

Reply:
One complication vs another. This way I'm passing only `args` through the abstraction layers, vs pass `input, args`. And template type is deduced anyway ;)
I'd prefer to leave it this way.

Reply:
Especially in kernels, input and arguments are separate entities. Recently I've been experimenting with relaxing the constraints on args (i.e. allowing Setup and Run to have different signatures), but the concept of "input" vs "configuration" remains quite stable. Here it's very clear - "window_length" determines the shape of the intermediate buffer, whereas other arguments only affect the contents of the result (not shape).

Reply:
I agree - Input is something you process, args are something that parametrize how your kernel behaves.